src/classes/window/fullscreen.vala

```diff
@@ -50,13 +50,20 @@ namespace pdfpc.Window {
         protected bool faded_to_black = false;
 
         /**
-         * Fixed layout
+         * Overlay layout. Holds all drawing layers (like the pdf,
+         * the pointer mode etc)
          */
-        protected Gtk.Fixed fixed_layout;
+        protected Gtk.Overlay overlay_layout;
 
-        public void add_to_fixed(Gtk.Widget w, int x, int y) {
-            this.fixed_layout.put(w, x, y);
-        }
+        /**
+         * Drawing area for pointer mode
+         */
+        public Gtk.DrawingArea drawing_surface { get; protected set; }
+
+        /**
+         * Video area for playback
+         */
+        public View.Video video_surface { get; protected set; }
```

Comment:
One minor comment: I didn't find it entirely clear that `video_surface` is a container for the actual videos when I first saw this, especially since you've subclassed `Gtk.Fixed` instead of using it directly. Adding the word "container" to the comment might help.